webapp/canonical_cla/views.py

```diff
@@ -0,0 +1,169 @@
+import base64
+import os
+from typing import Optional
+import urllib.parse as urlparse
+import requests
+import flask
+
+CANONICAL_CLA_API_URL = os.getenv("CANONICAL_CLA_API_URL")
+
+
+def get_query_param(url, key, is_base64=False) -> Optional[str]:
+    url_parts = list(urlparse.urlparse(url))
+    queries = dict(urlparse.parse_qs(url_parts[4]))
+
+    value = queries.get(key)
+    if value:
+        value = value[0] if isinstance(value, list) else value
+        if is_base64:
+            value = base64.b64decode(value).decode("utf-8")
+    return value
+
+
+def update_query_params(url: str, **params) -> str:
```

Comment:
these views are mainly proxies to the CLA API to have secure cookies access which is used for storing GitHub and Launchpad user access tokens.